engine/access/rpc/backend/backend_transactions.go

```diff
@@ -240,20 +240,75 @@ func (b *backendTransactions) GetTransactionResult(
 	}
 
 	// derive status of the transaction
-	status, err := b.deriveTransactionStatus(tx, transactionWasExecuted, block)
+	txStatus, err := b.deriveTransactionStatus(tx, transactionWasExecuted, block)
 	if err != nil {
 		return nil, convertStorageError(err)
 	}
 
 	return &access.TransactionResult{
-		Status:       status,
+		Status:       txStatus,
 		StatusCode:   uint(statusCode),
 		Events:       events,
 		ErrorMessage: txError,
 		BlockID:      blockID,
 	}, nil
 }
 
+func (b *backendTransactions) GetTransactionResultByIndex(
+	ctx context.Context,
+	blockID flow.Identifier,
+	index uint32,
+) (*access.TransactionResult, error) {
+	txID, err := b.transactions.TransactionIDByBlockIDIndex(blockID, index)
+	if err != nil {
+		return nil, convertStorageError(err)
+	}
+
+	tx, err := b.transactions.ByID(*txID)
+	if err != nil {
+		return nil, convertStorageError(err)
+	}
+
+	// create request and forward to EN
+	req := execproto.GetTransactionByIndexRequest{
```

Comment:
Hmm, don't we want to store this information locally? Otherwise with every query we have to hit EN

Reply:
This is unfortunately what we have to do right now, added a relevant TODO that we can circle back to